lib/fastlane/plugin/wpmreleasetoolkit/versioning/calculators/date_version_calculator.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'abstract_version_calculator'
+
+module Fastlane
+  module Wpmreleasetoolkit
+    module Versioning
+      # The `DateVersionCalculator` class is a specialized version calculator for date-based versions
+      # of an app, extending the `AbstractVersionCalculator` class.
+      class DateVersionCalculator < AbstractVersionCalculator
+        # Calculate the next date-based release version.
+        #
+        # If the current month is December, the method prompts the user to determine if the next
+        # release will be the first release of the next year. If so, it increments the major version
+        # and sets the minor version to 1, resetting the patch and build number components to zero.
+        # Otherwise, it calculates the next minor version.
```

Comment:
Is there any additional context about why we are doing this? I don't really understand why we'd want to bump a major version number on a date basis - rather than breaking changes or a major change.

At the very least, I'd expect the confirmation to clearly state that there will be a major version bump as otherwise this logic is buried deep in the implementation and not intuitive at all.

Maybe I am misunderstanding something?

Reply:
The Day One apps switched to a date-based versioning scheme at the beginning of 2023. So the first release of the year is 2023.1, second release is 2023.2, etc. Here's the discussion: pdrUDr-18L-p2 and pdrUDr-19B-p2

Reply:
But it seems [WPAndroid will also use this calculator](https://github.com/wordpress-mobile/WordPress-Android/pull/19288/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688R38) and it's not using the same versioning scheme. 😕

Reply:
Sorry @oguzkocer! That was my mistake. WPAndroid should be using the marketing version calculator. Updated here: https://github.com/wordpress-mobile/WordPress-Android/commit/4890c21d54f7936e5128bd275d9086fab2495984#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688

Reply:
Aaahh.. that makes sense, I'll take a look at it in my next review pass.

I promise this was not an elaborate way to say that this is a mistake. I was genuinely confused 😅

Reply:
@oguzkocer From the link you provided, WPAndroid will not use that `DateVersionCalculator` but instead use `MarketingVersionCalculator` (which bumps major version only after the minor version has reached `9`): https://github.com/wordpress-mobile/release-toolkit/pull/512/files/b3efb935ca85e84f8ccb8891c361e31cb52cac9b#diff-76d176e8878659afd4b838bca1da4227f9105be69d712c1346e1a9397a551020R8-R22